File: problem-4/main.lua
function Rev(n)
    local rev_num = 0
    while n > 0 do
        rev_num = rev_num * 10 + n % 10
        n = math.floor( n / 10 )
    end
    return rev_num
end

Largest = 0
for i = 100,999 do
    for j=100,999 do
        if i*j == Rev(i*j) and i*j > Largest then
            Largest = i*j
        end
    end
end
print(Largest)

File: problem-4/main.js
function rev(num) {
    var rev_num = 0;
    while (num > 0) {
        rev_num = rev_num * 10 + num % 10
        num = Math.floor(num / 10)
    }
    return rev_num
}

var largest = 0

for (let i = 100; i < 1000; i++) {
    for (let j = 100; j < 1000; j++) {
        if (i*j == rev(i*j) && i*j > largest) {
            largest = i*j
        }
    }  
}
console.log(largest)